### PR TITLE
NOOS-447/improve-ui-data-picker

### PR DESCRIPTION
**Pull request by zyassine**
**JIRA Reference**: [NOOS-447](https://noosenergy.atlassian.net/browse/NOOS-447)

[NOOS-447]: https://noosenergy.atlassian.net/browse/NOOS-447?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/inputs/InputText.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {TextField} from '@material-ui/core';
+
+const InputText = (props) => {
```
Comment (zyassine) [inline src/components/inputs/InputText.jsx:5]: @filiperosa description of InputText is missing

Comment (filiperosa) [reply src/components/inputs/InputText.jsx:23]: Thanks, will add it

**Comment by zyassine** [comment]
LGTM!

File: .circleci/config.yml [hunk 2]
```diff
@@ -68,7 +68,10 @@ jobs:
           tests_directory: ./tests
           <<: *poetry_install
       - store_test_results:
-          path: ".pytest_results"
+          path: '.pytest_results'
+      - run:
```
Comment (pieeri) [inline .circleci/config.yml:68]: 💡 

Maybe cleaner to add a specific step in the deployment pipeline with dedicated JS testing?

**Review by pieeri — CHANGES_REQUESTED** [review]
Small suggestion, but great improvement of the component library!

File: .circleci/config.yml (hunk 2 above)
Comment (filiperosa) [reply .circleci/config.yml:75]: Yes, added a new CI job for JS testing

File: .circleci/config.yml [hunk 3]
```diff
@@ -65,6 +65,13 @@ jobs:
...
+
+  test_package_javascript:
+    executor: noos-ci/default
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
       - run:
           name: JavaScript unit tests
           command: yarn install && yarn test
```
Comment (pieeri) [inline .circleci/config.yml:77]: 🐼 

JS dependencies and Dash JS components already created and attached into the CI workspace (`build_library` job) ; `yarn test` should be enough for such a job (and quicker).

**Review by pieeri — CHANGES_REQUESTED** [review]
🏓

File: .circleci/config.yml (hunk 3 above)
Comment (filiperosa) [reply .circleci/config.yml:77]: I tried before with `yarn test` alone and it seemed the node modules were not cached and it failed. I can try again.

Comment (filiperosa) [reply .circleci/config.yml]: Solved! Added `node_modules` folder to `persist_to_workspace` in the `build_library` job.
Cleaner and faster now! 👍

**Pull request merged**